lib/couchdb-client.go

```diff
@@ -418,10 +479,13 @@ func (c *CouchdbClient) Request(method string, uri string, body io.Reader) (resp
 type requestCountingRoundTripper struct {
 	RequestCount int
 	rt           http.RoundTripper
+	mutex        sync.Mutex
```

Comment:
Did you consider to use atomic counters like shown at https://gobyexample.com/atomic-counters instead of manually locking the counter variable? I'm also not the Golang expert per se, so I'm unsure about the implications of mutexes vs. atomic counters. I would prefer atomic counters, though, because the code would become more simple.

Reply:
I've made this change in the latest commit. I went ahead and changed all usages of the counter to be atomic.